benchmarks/matmul_benchmark/implementation.py

```diff
@@ -0,0 +1,70 @@
+from common.benchmark_wrapper import BenchmarkWrapper
+import json
+import os
+import torch
+import time
+
+
+class MatmulBenchmark(BenchmarkWrapper): #it's the same class inside benchmark_info.json
+
+    def __init__(self):
+        self.filePath = os.path.dirname(__file__) #os.path.dirname(__file__) returns the path of this file
+
+    def setSettings(self, settings_file):
+        """
+        Open and read the settings file
+        """
+        settings = os.path.join(self.filePath, "settings", settings_file)
+        self.dimension1 = json.load(open(settings, "r"))["dimension1"]
+        self.dimension2 = json.load(open(settings, "r"))["dimension2"]
+
+    def startBenchmark(self, verbosity=None):
+
+        print("The first matrix has dimensions", self.dimension1, "x", self.dimension2)
+        print("The second matrix has dimensions", self.dimension2, "x", self.dimension1)
+
+        start_time = time.time()
+
+        matrix_1 = torch.randn(self.dimension1, self.dimension2, device="cpu")
+        matrix_2 = torch.randn(self.dimension2, self.dimension1, device="cpu")
+
+        matrices_allocation_time = time.time() - start_time
+
+        print("The matrices has been allocated in %s s" % matrices_allocation_time)
+
+        product_cpu_start_time = time.time()
+
+        torch.matmul(matrix_1, matrix_2)

```

Comment:
Since we are writing GPU benchmarks I would avoid falling back to CPU when no GPU is available... I prefer the sad smiley!
I'd let @leggerf comment on this, but I think this can become tricky to say the least.

Reply:
I would fallback to CPU, but this should be clear in the printout. This will also allow to test the code functionalities on devices without CPU. But I'm fine with whatever choice, as long as the code doesn't crash but exits cleanly